rplidar.go

```diff
@@ -335,7 +335,7 @@ func (d *Device) NextPointCloud(ctx context.Context) (pointcloud.PointCloud, err
 	defer d.mu.Unlock()
 	pc, timeStamp, err := d.getPointCloud(ctx)
 	if err != nil {
-		return nil, err
+		return nil, nil//err
```

Comment:
why did you change this?

Reply:
Since we'll be pinging pointcloud a lot and frequently there is a chance that the rplidar will error out a particular call. This allows us to ignore that missed point cloud and continue without saving it.

Reply:
I see. If that's the desired behavior, then I'd say you should do this:
1) Make that happen in the file that calls this function, e.g. in a file like `cmd/savepcdfiles/main.go`
2) Don't do this preemptively, though. I haven't had any issues with this yet, so I don't see a reason to incorporate this (yet). If it happens to become an issue, though, then we'll have to find a smarter solution than simply ignore an error, e.g. at the very least print out a warning.

Reply:
Revert this back to the way it was.

Reply:
Yes, when starting up it would error out due to rplidar c++ stuff. And I believe I also saw this happen several times after ~1 in 20 polls. and that was at 200 ms intervals

Reply:
A few thoughts here:

Returning nil with no error is not a good idea, especially for a function that's part of the Camera{} interface itself. Someone writing a consumer for that API is going to check for an error, see it's "all good" when there is none, then have their program crash because it tries to reference a nil pointer.

This is also a pretty high level place to ignore all errors.

Since this function actually does a bunch of things, I'd suggest you find lowest level you can where the actual error is being generated and then filter errors there. That is, actually check that the error being generated is specifically the one that you know is "normal" to ignore, and only ignore that one. You don't want to miss if some other type of error happens.

Secondly, and possibly more useful, is when that happens, have it retry whatever function just failed, so that calls still ultimately succeed.

Ex: I'm working on this industrial camera code right now, and when I pull an image buffer from it, it can be "incomplete" so in that case, I just skip it and and pull the next, and the consumer code at the end never sees a problem. But any other sorts of errors can still bubble up to the consumer API call.